rest-server/engines/src/main/scala/com/actionml/engines/navhinting/NavHintingAlgorithm.scala

```diff
@@ -76,10 +75,21 @@ class NavHintingAlgorithm(dataset: NavHintingDataset)
     val converted = datum.event.properties.flatMap(_.conversion).getOrElse(false)
     if (converted) { // update the model with the active journeys and remove it from active
       if (activeJourney.nonEmpty) { // have an active journey
-        updateModel(Journey(datum.event.entityId, activeJourney.get), OffsetDateTime.parse(datum.event.eventTime))
+        // create and empty model if this is a new cconversion-id or find the right navHintModel
+        val navHintsModel = if(navHintsModels.contains(datum.event.targetEntityId)) navHintsModels(datum.event.targetEntityId) else {
+          // no model yet so create one and add it to the list
+          // always persist the new model's id after creating it!
+          val model = MongoStorage.getStorage(engineId, MongoStorageHelper.codecs)
```

Comment:
`MongoStorage.getStorage` is expensive. It's better to have Storage instance as NHAlgorithm member